File: 이분 탐색(결정알고리즘) & 그리디 알고리즘/이분 탐색.py
import sys
# sys.stdin = open("input.txt", "rt")

"""
n, m = map(int, input().split())
arr = list(map(int, input().split()))
arr.sort()

def binary_search(start, end, target):
    if start > end:
        return False
    pivot = (start + end) // 2
    if arr[pivot] > target:
        return binary_search(start, pivot - 1, target)
    elif arr[pivot] < target:
        return binary_search(pivot + 1, end, target)
    else:
        return pivot + 1

print(binary_search(0, n - 1, m))
"""

n, m = map(int, input().split())
arr = list(map(int, input().split()))
arr.sort()

left = 0
right = n - 1
while left <= right:
    pivot = (left + right) // 2
    if arr[pivot] < m:
        left = pivot + 1
    elif arr[pivot] > m:
        right = pivot - 1
    else:
        print(pivot + 1)
        break

File: 정렬 & 이분탐색(결정알고리즘) & 투포인트 알고리즘 & 스택/이분탐색.cpp
#include <stdio.h>
#include <vector>
#include <algorithm>
using namespace std;

int main(void){
	// freopen("input.txt", "r", stdin);
	int n, m, i, left, right, mid;
	scanf("%d%d", &n, &m);
	vector<int> arr(n);
	
	for(i = 0; i < n; i++){
		scanf("%d", &arr[i]);	
	}
	sort(arr.begin(), arr.end());
	
	left = 0;
	right = n - 1;
	while(left <= right){
		mid = (left + right) / 2;
		if(arr[mid] == m){
			printf("%d", mid + 1);
			break;
		}
		else if(arr[mid] > m) right = mid - 1;
		else 				  left = mid + 1;
	}
	
	return 0;	
}
